Fix tests for eigenstates, stochastic and floquet solvers

### Bug Description

During the last month, the automated tests ran ~500 times and we had 6 random failures:

`qutip/tests/core/test_eigenstates.py::test_satisfy_eigenvalue_equation[degenerate-dense-True]` - failed 3 times
`qutip/tests/solver/test_stochastic.py::test_run_from_experiment_open[True-euler]` - failed twice
`qutip/tests/solver/test_floquet.py::test_fsesolve_fallback` - failed once

We should fix them. _Possible solution_: "loosen the tolerances or change the input generation (size, density, etc.) until the failure rate is acceptable, without making it so tolerant that anything would pass"

### Code to Reproduce the Bug

No related code to reproduce it - due to random nature of tests, the above-mentioned errors can be observed _occasionally_ in CI/CD tests and is not related to code changes of a PR.

### Code Output

From _[ubuntu-latest, python3.11, p311, numpy fallback, no mkl]_ and _[ubuntu-latest, python3.12, p312, numpy fallback]_:

```shell
FAILED qutip/tests/solver/test_stochastic.py::test_run_from_experiment_open[False-euler] - RuntimeWarning: overflow encountered in square
```

From _[ubuntu-latest, python3.14, python 3.14 - Latest]_:

```shell
FAILED qutip/tests/solver/test_floquet.py::TestFloquet::testFloquetUnitary - assert (0.9999981730...34135176e-05j) == 1.0 ± 5.0e-05
  comparison failed
  Obtained: (0.9999981730805094+5.2031557134135176e-05j)
  Expected: 1.0 ± 5.0e-05
FAILED qutip/tests/solver/test_stochastic.py::test_run_from_experiment_open[False-euler] - RuntimeWarning: overflow encountered in square
```

### Expected Behaviour

Tests don't fail anymore

### Your Environment

```shell
Dependencies installed in CI/CD jobs w.r.t. to platform and Python version
```

### Additional Context

Based on the discussion in #1268

--- a/qutip/tests/core/test_eigenstates.py
+++ b/qutip/tests/core/test_eigenstates.py
@@ -78,7 +78,7 @@ def random_hamiltonian(request):
     eigen = None
     dist = "fill"
     if dimensions == [3, 3, 3]:
-        eigen = [0, 1, 1] * 9
+        eigen = [1, 2, 3] * 9
         dist = "eigen"
     return qutip.rand_herm(dimensions, distribution=dist, eigenvalues=eigen)
 

--- a/qutip/tests/solver/test_stochastic.py
+++ b/qutip/tests/solver/test_stochastic.py
@@ -464,7 +464,7 @@ def test_run_from_experiment_open(method, heterodyne):
     a = destroy(N)
     sc_ops = [a, a.dag() * 0.1]
     psi0 = basis(N, N-1)
-    tlist = np.linspace(0, 1, 251)
+    tlist = np.linspace(0, 0.5, 501)
     options = {
         "store_measurement": "start",
         "dt": tlist[1],

--- a/qutip/tests/test_utilities.py
+++ b/qutip/tests/test_utilities.py
@@ -125,25 +125,29 @@ def test_cpu_count(monkeypatch):
 
 def test_zeta():
     mpmath = pytest.importorskip("mpmath")
-    N = 100
+    N = 50
     s_list = np.random.rand(N) + 1.0001
     # could randomly fail if q is an integer <=0
-    q_list = (np.random.rand(N) - 0.5) * 10
+    q_list = (np.random.rand(N) + 0.05) * 10
     for s, q in zip(s_list, q_list):
-        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-14)
+        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-12)
 
-    s_list = np.random.rand(N) + 1.0001
+    q_list = (np.random.rand(N) * 0.9 - .95 - np.random.randint(10))
+    for s, q in zip(s_list, q_list):
+        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-12)
+
+    s_list = np.random.rand(N) + 1.001
     q_list = (np.random.rand(N) - 0.5) * 10 + np.random.randn(N) * 10j
     for s, q in zip(s_list, q_list):
-        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-14)
+        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-12)
 
     s_list = (
         1 + (np.random.lognormal(size=N) + 1e-6)
-        * np.exp((np.random.rand(N) - 0.5) * 1j * np.pi)
+        * np.exp((np.random.rand(N) * 0.8 - 0.4) * 1j * np.pi)
     )  # Real part > 1.000001
-    q_list = (np.random.rand(N) - 0.5) * 10 + np.random.randn(N) * 10j
+    q_list = (np.random.rand(N) + 0.1) * 10 + np.random.randn(N) * 10j
     for s, q in zip(s_list, q_list):
-        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-14)
+        assert utils.zeta(s, q) == pytest.approx(mpmath.zeta(s, q), rel=1e-12)
 
 
 class TestFitting:

